Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -103,19 +246,18 @@ void CookieStore::get(CookieStoreGetOptions&& options, Ref<DeferredPromise>&& pr
         return;
     }
 
-    if (!m_cookieJar) {
-        promise->reject(SecurityError);
-        return;
-    }
+    m_promises.add(++m_nextPromiseIdentifier, WTFMove(promise));
+    auto completionHandler = [promiseIdentifier = m_nextPromiseIdentifier](CookieStore& cookieStore, ExceptionOr<Vector<Cookie>>&& result) {
+        auto promise = cookieStore.takePromise(promiseIdentifier);
+        if (!promise)
+            return;
 
-    auto& document = *downcast<Document>(context);
-    auto completionHandler = [promise = WTFMove(promise)] (std::optional<Vector<Cookie>>&& cookies) {
-        if (!cookies) {
-            promise->reject(TypeError);
+        if (result.hasException()) {
+            promise->reject(result.releaseException());
             return;
         }
 
-        auto& cookiesVector = *cookies;
```

Comment:
We don't need to `releaseReturnValue()` here. You can just do this like we used to:
```
auto& cookies = result.returnValue();
```

Also renaming cookiesVector to cookies would be nice now that there is no longer a naming conflict.

Reply:
Should I also change `.releaseException' to just `.exception()`?

Reply:
`.releaseException()` since reject() takes in an `Exception` so you want an r-value reference to call the Exception's move constructor instead of doing a copy.

Reply:
If I have the cookies vector as a reference, then I'm not able to construct the `CookieListItem` below by doing `WTFMove(cookies[0])`

Reply:
Keeping releaseReturnValue for now.